garak/generators/huggingface.py

```diff
@@ -492,14 +516,27 @@ def _call_model(
         if self.top_k is not None:
             self.generation_config.top_k = self.top_k
 
-        text_output = []
+        raw_text_output = []
         with warnings.catch_warnings():
             warnings.simplefilter("ignore", category=UserWarning)
             with torch.no_grad():
+                if self.use_chat:
+                    formatted_prompt = self.tokenizer.apply_chat_template(
+                        self._format_chat_prompt(prompt),
+                        tokenize=False,
```

Comment:
reading the HF docs this recommended use 'tokenize' smells of insufficient encapsulation, but lgtm for our implementation